### DISCUSSION TITLE
Make it easier to add/remove tags

### DISCUSSION THREAD

**Discussion by fivestones** [mealie-recipes/mealie#2119] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Currently, in order to add or remove tags, I have to open the recipe, click edit, scroll down to where the tags are, add/remove the tags, scroll back up to the save button, and click it.

**(Optional): Suggest A Solution**
It should be possible to tag things from either the recipe view, or even from the list of recipes. In the recipe view, there should be a text box where the current tags are listed, and I should be able to just start typing there and it should auto-complete with existing matching tags which I can pick from, or just finish typing a new tag; either way after typing/picking the tag I want, I should be able to just hit enter, and the recipe will be tagged. Current tags should have an "x" next to them when I hover over them, which will delete that tag when I click it.

From the "all recipes" view or a search results view, I should be able to click the "..." button and have a way to similarly edit tags right there.

In addition, there should be a way in the "all recipes" view or a search results view to select multiple recipes and then mass tag them all with any given tag or tags.

**Additional Information**
_FILL THESE OUT_
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

Thanks for making this great software!

**Comment by RealFoxie** [comment]
For bulk editing of tags, did you check out /group/data/recipes? I think that might already provide what you need in that regard

**Reply by Cereal916** [reply]
Amazing, this is exactly what I needed. This is a great hidden tool, is there any other way to get there without manually editing the url?

**Comment by fivestones** [comment]
Oh! Thanks for this. I had never seen that page. It does indeed look like it should do what I was wanting.

How is this discoverable? I don't see any link to this page. Is it only available with the direct link? If so is that because it is 'an experimental feature'?

Also, at least when I just tried, tagging works fine to bulk tag things, but setting a category seems to just pull up the tags, not the categories. Maybe I should file a bug report about that? I'll take a look to see if someone already has.

**Reply by RealFoxie** [reply]
To find the page:
Click on your own name (top left corner), scroll down to manage data, then choose "recipes" from the dropdown (which is set on "food" by default).
I haven't used the bulk things, but I just tested it and tags and categories do give the same dropdown, so I think you are right.
Filing a bug report is a good idea 👍

**Reply by michael-genson** [reply]
FYI the bug report: https://github.com/mealie-recipes/mealie/issues/2200

**Comment by FancyNerd92** [comment]
I have problem with automatic tags from recipe on bulk import. in single mode you have the option to use "import original keywords as tags" but in bulk mode you don't have this option. So in my case more than 2K recipies must upload url by url to have this option? I want it because i use it for automatic ingridient tag on my recipes but i can't use it on bulk import... Please give me a clue! thanks!

**Comment by henricook** [comment]
My 10th visit to this issue to remind myself how to edit tags 😅
